code/amcss.c

```diff
@@ -128,6 +128,24 @@ static void test_stepper(mps_addr_t object, mps_fmt_t fmt, mps_pool_t pool,
 }
 
 
+/* area_scan -- area scanning function for mps_pool_walk */
+
+static mps_res_t area_scan(mps_ss_t ss, void *base, void *limit, void *closure)
+{
+  unsigned long *count = closure;
+  mps_res_t res;
+  while (base < limit) {
+    mps_addr_t prev = base;
```

Comment:
Here and elsewhere I see void * and mps_addr_t used interchangeably, is there a reason to use void* when we have mps_addr_t? This assumes mps_addr_t will never change its underlying type (which I guess is a fairly safe assumption)

Reply:
The two types are used to make a distinction: `mps_addr_t` represents an address in memory that is managed by the MPS ([see the manual](https://www.ravenbrook.com/project/mps/master/manual/html/topic/interface.html#c.mps_addr_t)), while `void *` represents any address, that might or might not be in memory managed by the MPS.

In the case of amcss.c, `area_scan()` has to belong to the type `mps_area_scan_t` (so that its arguments are `void *`). Area scanning functions were introduced by @rptb1 to fix [job003668](https://www.ravenbrook.com/project/mps/issue/job003668/), but I am not sure why he chose to use `void *` for the area scanning arguments and not `mps_addr_t`.

@rptb1 Do you remember why you made this choice?